File: 3sem/poo/Aula01-Codigo/C++/Exemplo02-Pessoa/Pessoa.cpp
#include <string>
#include <iostream>

using namespace std;

class Pessoa {
 public:
    string nome;
    string DataNasc;
    string cor;
    bool bDormindo;

    Pessoa(string umNome, string umaDataNasc, string umaCor) {
        nome = umNome;
        DataNasc = umaDataNasc;
        cor = umaCor;
        bDormindo = false;
    };

    void Acordar(){
        if(!bDormindo){
            cout << "Ja estou acordado" << "\n";
        }else{
            bDormindo = false;
            cout << "Acordei, eba" << "\n";
        };
    };
    void Dormir(){
        if(bDormindo){
            cout << "Ja estou dormindo" << "\n";
        } else {
            bDormindo = true;
            cout << "Dormi, zzz" << "\n";
        };
    };
};

File: 3sem/poo/Aula01-Codigo/Java/Exemplo02-Pessoa/src/exemplo2/Pessoa.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package exemplo2;

/**
 *
 * @author Junio
 */
public class Pessoa {
    String nome;
    String DataNasc;
    String cor;
    boolean bDormindo;

    public Pessoa(String nome, String DataNasc, String cor) {
        nome = nome;
        DataNasc = DataNasc;
        cor = cor;
        bDormindo = false;
    }

    public void Acordar(){
        if(!bDormindo){
            System.out.println("Ja estou acordado");
        }else{
            bDormindo = false;
            System.out.println("Acordei, eba");
        }
    }
    public void Dormir(){
        if(bDormindo){
            System.out.println("Ja estou dormindo");
        } else {
            bDormindo = true;
            System.out.println("Dormi, zzz");
        }
    }
}
